src/Table/Table.tsx

```diff
@@ -0,0 +1,984 @@
+                        props.primaryActionButton.onClick(e, props.selectedRow);
+                    }}
+                >
+                    {props.primaryActionButton.label}
+                </button>
+            }
+            {props.actionLinks && props.actionLinks.length > 0 &&
+                <div
+                    className="ellipsis-dropdown-holder"
+                    onClick={(e) => {
+                        const tableSize: DOMRect = props.tableRef?.current?.getBoundingClientRect();
+                        const actionColumnHeight: number = actionRef.current?.scrollHeight;
+                        const actionColumSize: DOMRect = actionRef.current?.getBoundingClientRect();
+
+                        if (tableSize?.height > actionColumnHeight) {
+                            const lengthOffset: number = tableSize?.bottom - actionColumSize?.bottom;
+                            if (lengthOffset < actionColumnHeight) {
+                                setDropup(true);
+                            } else {
+                                setDropup(false);
+                            }
+                        } else {
+                            setDropup(false);
+                        }
+                        e.preventDefault();
+                        props.onActionDropped(e);
+                    }}
+                >
+                    <div className="icon-holder" id={"ellipsis-" + props.selectedRow.rowIndex}>
+                        {ellipsis}
+                    </div>
+                    <div
+                        className={getActionColumnClasses()}
+                        ref={actionRef}
+                    >
+                        {props.actionLinks.map((link: ActionLinkItem, index: number) =>
+                            <a
+                                href="#"
+                                key={`${link.label.replace(" ", "-")} - ${index}}`}
+                                onClick={(e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+                                    e.preventDefault();
+                                    link.onClick(e, props.selectedRow);
+                                }}
+                            >
+                                {link.label}
+                            </a>
+                        )}
+                    </div>
+                </div>
+            }
+        </div>
+    );
+};
+
+interface TableUIProps {
+    actionLinks?: Array<ActionLinkItem>;
+    allRowsAreSelected?: boolean;
+    className: string;
+    columns: Array<TableHeader>;
+    footer: React.ReactNode;
+    loading: boolean;
+    onActionDropped: (event: React.MouseEvent<HTMLDivElement, MouseEvent>, row: TableRow, rowIndex?: number) => void;
+    onAllItemsSelected?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    onItemSelected?: (e: React.ChangeEvent<HTMLInputElement>, row: TableRow, type: "row" | "subRow", rowIndex?: number) => void;
```

Comment:
type: "row" | "subRow" should be created as a type and used across the component